tests/SmartRange.test.ts

```diff
@@ -0,0 +1,195 @@
+import { describe, expect, test } from "vitest";
+import { SmartRange } from "../src";
+
+describe("SmartRange", () => {
```

Comment:
Maybe a test with a really big SmartRange could be interesing for performance tests

Reply:
Maybe, but I'm afraid 😜
Jk will do.

Reply:
I decided to leave it out for now, while indexing a "large" (>10e6) SmartRange object is fast, converting it too an array slows the tests down too much (>2sec) for my taste 😅